Mathlib/NumberTheory/Harmonic/ZetaAsymp.lean

```diff
@@ -347,4 +352,71 @@ lemma _root_.isBigO_riemannZeta_sub_one_div {F : Type*} [Norm F] [One F] [NormOn
 
 end continuity
 
+section val_at_one
+
+open Complex
+
+lemma tendsto_Gamma_term_aux : Tendsto (fun s ↦ 1 / (s - 1) - 1 / Gammaℝ s / (s - 1)) (𝓝[≠] 1)
+    (𝓝 (-(γ + Complex.log (4 * ↑π)) / 2)) := by
+  have h := hasDerivAt_Gammaℝ_one
+  rw [hasDerivAt_iff_tendsto_slope, slope_fun_def_field, Gammaℝ_one] at h
+  have := h.div (hasDerivAt_Gammaℝ_one.continuousAt.tendsto.mono_left nhdsWithin_le_nhds)
+    (by rw [Gammaℝ_one]; exact one_ne_zero)
+  rw [Gammaℝ_one, div_one] at this
+  refine this.congr' ?_
+  have : {z | 0 < re z} ∈ 𝓝 (1 : ℂ) := by
+    apply (continuous_re.isOpen_preimage _ isOpen_Ioi).mem_nhds
+    simp only [mem_preimage, one_re, mem_Ioi, zero_lt_one]
+  rw [EventuallyEq, eventually_nhdsWithin_iff]
+  filter_upwards [this] with a ha _
+  rw [Pi.div_apply, ← sub_div, div_right_comm, sub_div' _ _ _ (Gammaℝ_ne_zero_of_re_pos ha),
+    one_mul]
+
+lemma tendsto_riemannZeta_sub_one_div_Gammaℝ :
+    Tendsto (fun s ↦ riemannZeta s - 1 / Gammaℝ s / (s - 1)) (𝓝[≠] 1)
+    (𝓝 ((γ - Complex.log (4 * ↑π)) / 2)) := by
+  have := tendsto_riemannZeta_sub_one_div.add tendsto_Gamma_term_aux
+  simp_rw [sub_add_sub_cancel] at this
+  convert this using 1
+  ring_nf
+
+/-- Formula for `ζ 1`. Note that mathematically `ζ 1` is undefined, but our construction ascribes
+this particular value to it. -/
+lemma _root_.riemannZeta_one : riemannZeta 1 = (γ - Complex.log (4 * ↑π)) / 2 := by
+  have := (HurwitzZeta.tendsto_hurwitzZetaEven_sub_one_div_nhds_one 0).mono_left
+    <| nhdsWithin_le_nhds (s := {1}ᶜ)
+  simp only [HurwitzZeta.hurwitzZetaEven_zero, div_right_comm _ _ (Gammaℝ _)] at this
+  exact tendsto_nhds_unique this tendsto_riemannZeta_sub_one_div_Gammaℝ
+
+/-- Formula for `Λ 1`. Note that mathematically `Λ 1` is undefined, but our construction ascribes
+this particular value to it. -/
+lemma _root_.completedRiemannZeta_one :
+    completedRiemannZeta 1 = (γ - Complex.log (4 * ↑π)) / 2 :=
+  (riemannZeta_one ▸ div_one (_ : ℂ) ▸ Gammaℝ_one ▸ riemannZeta_def_of_ne_zero one_ne_zero).symm
+
+/-- Formula for `Λ₀ 1`, where `Λ₀` is the entire function satisfying
+`Λ₀ s = π ^ (-s / 2) Γ(s / 2) ζ(s) + 1 / s + 1 / (1 - s)` away from `s = 0, 1`.
+
+Note that `s = 1` is _not_ a pole of `Λ₀`, so this statement (unlike `riemannZeta_one`) is
+a mathematically meaningful statement and is not dependent on Mathlib's particular conventions for
+division by zero. -/
+lemma _root_.completedRiemannZeta₀_one :
+    completedRiemannZeta₀ 1 = (γ - Complex.log (4 * ↑π)) / 2 + 1 := by
+  have := completedRiemannZeta_eq 1
+  rw [sub_self, div_zero, div_one, sub_zero, eq_sub_iff_add_eq] at this
+  rw [← this, completedRiemannZeta_one]
+
+/-- With Mathlib's particular conventions, we have `ζ 1 ≠ 0`. -/
+lemma _root_.riemannZeta_one_ne_zero : riemannZeta 1 ≠ 0 := by
+  -- This one's for you, Kevin.
+  rw [riemannZeta_one, ← ofReal_ofNat, ← ofReal_mul, ← ofReal_log (by positivity),
+    ← ofReal_sub, ← ofReal_ofNat, ← ofReal_div, ofReal_ne_zero]
```

Comment:
This looks like `norm_cast` should be able to do it (after `rw [riemannZeta_one]`).
I think the problem is `ofReal_log` that needs a positivity assumption (`ofReal_log` is also not tagged `norm_cast`, but trying to add the attribute locally does not help). It would be nice if `norm_cast` could do this, but this is another topic...

Reply:
I think it is a bit clearer than doing all the explicit rewrites with `ofReal` lemmas, so yes.

Reply:
Gesagt, getan.